tests/pep8-git-pr-checker

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+from __future__ import with_statement
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+# 501: line too long
+ignore_codes = ['E501']
+
+def system(*args, **kwargs):
+    kwargs.setdefault('stdout', subprocess.PIPE)
+    proc = subprocess.Popen(args, **kwargs)
+    out, err = proc.communicate()
+    return out
+
+
+def main():
+    files = []
+    output = system('git', 'ls-files')
+    for tmp_file in output.split():
+        if tmp_file.endswith('.py'):
+            files.append(tmp_file)
+
+    tempdir = tempfile.mkdtemp()
+    for name in files:
+        filename = os.path.join(tempdir, name)
+        filepath = os.path.dirname(filename)
+        if not os.path.exists(filepath):
+            os.makedirs(filepath)
+        with file(filename, 'w') as f:
+            system('git', 'show', ':' + name, stdout=f)
+
+    args = ['pep8']
+    args.extend(('--ignore', ','.join(ignore_codes)))
+    args.append('.')
+    output = system(*args, cwd=tempdir)
+    shutil.rmtree(tempdir)
+    if output:
+        print(output)
+        sys.exit(1)
```

Comment:
we use travis-ci to run pep8 check for each PR. if it detects any issues, exit is called to mark as the PR has failed pep8 check. Ex: https://travis-ci.org/resin-io/resin-sdk-python